### PR TITLE
Ned/add testing

### PR DESCRIPTION
**Pull request by nedbat**
Testing is cool.  @cpennington I'm sure you agree.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+default: test
+
+test:
+	python test/test_func.py
+
+clean:
+	-rm -rf *.egg-info
+	-rm -f *.pyc */*.pyc */*/*.pyc */*/*/*.pyc */*/*/*/*.pyc */*/*/*/*/*.pyc
+	-rm -rf __pycache__ */__pycache__ */*/__pycache__ */*/*/__pycache__ */*/*/*/__pycache__ */*/*/*/*/__pycache__
+	-rm -f MANIFEST
```
Comment (cpennington) [inline Makefile]: What does the `-` prefix mean?

**Comment by cpennington** [comment]
I'm a bit baffled by how/why this works, but you appear to be invoking magic from the pylint test suite, so I'm happy to assume it's ok. We should hook up to travis, though.

File: Makefile (hunk 1 above)
Comment (nedbat) [inline Makefile]: Ignore errors: https://www.gnu.org/software/make/manual/html_node/Errors.html#Errors

**Comment by nedbat** [comment]
TBH, this is mostly copy-paste from other test suites.  I looked a bit at the Pylint test runner.  The name of the .py file determines a number of things, including whether to expect pylint violations, and what versions of Python it applies to (though none of my tests use versions).  If the file is called "func_noerror_*.py", then we expect no violations.  Otherwise, the file is expected to generate the violations found in the same-named file in the messages folder, replacing ".py" with ".txt"

**Comment by cpennington** [comment]
:

**Comment by cpennington** [comment]
:+1:

**Pull request merged**